pkg/metrics/roundtripper.go

```diff
@@ -0,0 +1,151 @@
+package metrics
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/operator-framework/operator-marketplace/pkg/defaults"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// appRegistryRoundTripperCounter wraps a RoundTripperFunc with functions that record prometheus metrics
+// when reaching out to an AppRegistry.
+func appRegistryRoundTripperCounter(counter *prometheus.CounterVec, next http.RoundTripper) promhttp.RoundTripperFunc {
+	return promhttp.RoundTripperFunc(func(r *http.Request) (*http.Response, error) {
+		resp, err := next.RoundTrip(r)
+		if err == nil {
+			if quayNamespace, ok := r.URL.Query()["namespace"]; ok && len(quayNamespace) == 1 {
```

Comment:
nit: encapsulate the logic to get the name of quay namespace from a request into a function.